moduleframework/tools/modulelint.py

```diff
@@ -43,57 +43,75 @@ def setUp(self):
         if self.dp.dockerfile is None:
             self.skip()
 
-    def testArchitectureInEnvAndLabelExists(self):
+    def test_architecture_in_env_and_label_exists(self):
         self.assertTrue(self.dp.get_docker_specific_env("ARCH="))
         self.assertTrue(self.dp.get_specific_label("architecture"))
 
-    def testNameInEnvAndLabelExists(self):
+    def test_name_in_env_and_label_exists(self):
         self.assertTrue(self.dp.get_docker_specific_env("NAME="))
         self.assertTrue(self.dp.get_specific_label("name"))
 
-    def testReleaseLabelExists(self):
+    def test_release_label_xists(self):
         self.assertTrue(self.dp.get_specific_label("release"))
 
-    def testVersionLabelExists(self):
+    def test_version_label_exists(self):
         self.assertTrue(self.dp.get_specific_label("version"))
 
-    def testComRedHatComponentLabelExists(self):
+    def test_com_redhat_component_label_exists(self):
         self.assertTrue(self.dp.get_specific_label("com.redhat.component"))
 
-    def testIok8sDescriptionExists(self):
-        self.assertTrue(self.dp.get_specific_label("io.k8s.description"))
 
-    def testIoOpenshiftExposeServicesExists(self):
-        label_io_openshift = "io.openshift.expose-services"
-        exposes = self.dp.get_docker_expose()
-        label_list = self.dp.get_docker_labels()
-        self.assertTrue(label_list[label_io_openshift])
-        for exp in exposes:
-            self.assertTrue("%s" % exp in label_list[label_io_openshift])
+class DockerfileLinterInContainer(container_avocado_test.ContainerAvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    def test_docker_nodocs(self):
+        self.start()
+        installed_pkgs = self.run("rpm -qa --qf '%{{NAME}}\n'", verbose=False).stdout
```

Comment:
just replace by `test -e $(rpm -qad)`

Reply:
This is questionable whether we would like to test only packages delivered by module or the whole system including Platform.
For some cases this would be a tricky and we need to have a whitelist.
Personally, I don't want to have this test personally failing because of e.g Platform will contain some documents which will be needed. What do you think? Does it make sense?

Reply:
no, there should be no whitelist. if there are docs installed, this container is not following guidelines. whether docs are coming from different modules doesn't matter because we are testing single unit.

Reply:
I think this test is a good candidate for a warning and not an error. To support Petr, here is a brief analysis:

```
$ docker run -ti registry.fedoraproject.org/fedora-minimal:26 bash
bash-4.4# rpm -qad | wc -l
544

$ docker run -ti docker.io/jamesantill/boltron-27 bash
Boltron-27-2ddd828be4c9-0# rpm -qad | wc -l
5529

$ docker run -ti registry.access.redhat.com/rhel7 bash
[root@025a6e693f15 /]# rpm -qad | wc -l
1951
```

Igor, I agree with you that it would be great to test complete images and have
them valid. But as you can see, all our base images contain tons of
documentation. Therefore maybe we could have 2 use cases covered here (by 2
different tests):
- `rpm -qad` which would produce a warning in case of failure
- `rpm -qd $PACKAGES_INSTALLED_IN_THIS_IMAGE` which could produce an error

WDYT?

Reply:
I think that you are doing wrong thing. RPM will always show you docs, but you don't test files for existence.

Reply:
wow, haven't realized that, good point Igor! new findings then:

```
$ docker run -ti registry.fedoraproject.org/fedora-minimal:26 bash
bash-4.4# for f in $(rpm -qad); do test -e $f && echo $f; done
/usr/share/info/dir

$ docker run -ti docker.io/jamesantill/boltron-27 bash                                                                                                         1 ↵
Boltron-27-d12f22f5458d-0# for f in $(rpm -qad); do test -e $f && echo $f; done | wc -l
4037

$ docker run -ti registry.access.redhat.com/rhel7 bash
[root@23e1d08c9da0 /]# for f in $(rpm -qad); do test -e $f && echo $f; done
/usr/share/info/dir

[root@23e1d08c9da0 /]# rpm -qf /usr/share/info/dir
info-6.5-1.fc28.x86_64

[root@23e1d08c9da0 /]# cat /usr/share/info/dir
This is the file .../info/dir, which contains the
topmost node of the Info hierarchy, called (dir)Top.
The first time you invoke Info you start off looking at this node.

File: dir,      Node: Top       This is the top of the INFO tree

  This (the Directory node) gives a menu of major topics.
  Typing "q" exits, "?" lists all Info commands, "d" returns here,
  "h" gives a primer for first-timers,
  "mEmacs<Return>" visits the Emacs manual, etc.

  In Emacs, you can click mouse button 2 on a menu item or cross reference
  to select it.

* Menu:
```

who will fix `info` package?

Reply:
This is probably bug in RPM rather than in `texinfo` package...

```
%config(noreplace) %verify(not md5 size mtime) %{_infodir}/dir
```

I don't see why it would be marked as %doc. Can you report bug on https://github.com/rpm-software-management/rpm ?